conans/client/cmd/export.py

```diff
@@ -447,41 +448,51 @@ def _export_scm(scm_data, origin_folder, scm_sources_folder, output):
 
 
 def export_source(conanfile, origin_folder, destination_source_folder):
-    if isinstance(conanfile.exports_sources, str):
-        conanfile.exports_sources = (conanfile.exports_sources, )
+    if callable(conanfile.exports_sources):
+        conanfile.deps_cpp_info = DepsCppInfo()
+        conanfile.copy = FileCopier([origin_folder], destination_source_folder)
+        conanfile.export_sources()
```

Comment:
PLease use:

```python
conan_file.output.highlight("Running export_sources()")
with conanfile_exception_formatter(str(conan_file), "export_sources"):
```
to wrap it and provide better user messages in case of exceptions